test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/ControllerTest.cs

```diff
@@ -274,6 +276,48 @@ public void TempData_CanSetAndGetValues()
             Assert.Equal(input, result);
         }
 
+        public static TheoryData<object, Type> IncompatibleModelData
+        {
+            get
+            {
+                // Small grab bag of instances and expected types with no common base except typeof(object).
+                return new TheoryData<object, Type>
+                {
+                    { null, typeof(object) },
+                    { true, typeof(bool) },
+                    { 43.78, typeof(double) },
+                    { "test string", typeof(string) },
+                    { new List<int>(), typeof(List<int>) },
+                    { new List<string>(), typeof(List<string>) },
+                };
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(IncompatibleModelData))]
+        public void ViewDataModelSetter_DoesNotThrow(object model, Type expectedType)
+        {
+            // Arrange
+            var activator = new ViewDataDictionaryControllerPropertyActivator(new EmptyModelMetadataProvider());
```

Comment:
:see_no_evil: because "test 1" determining the type isn't meaningful -- the type is `ViewDataDictionary` and not `ViewDataDictionary<object>`. IOW would need to expose `_declaredModelType` to write test 1 alone. So I'm leaving this as an integration test.